aiida/orm/nodes/data/code.py

```diff
@@ -323,6 +323,19 @@ def get_input_plugin_name(self):
         """
         return self.get_attribute('input_plugin', None)
 
+    def set_withmpi(self, withmpi):
+        """
+        Set whether a code is compiled with mpi support or not, to be used
+        for the automatic generation of a new calculation.
+        """
+        self.set_attribute('withmpi', bool(withmpi))
```

Comment:
Should really add a type check here and raise `TypeError` if incorrect. Something like:
```python
from aiida.common.lang import type_check
type_check(withmpi, bool)
```

Reply:
done.